Remove the first false hardware interruption

ypflll opened the pull request:
In radxa.c, function "poll" is called before "lseek" and it will output a false interruption at the beginning, even there is no interruption source. The same problem should happen on other boards. This is only tested on radxaboard.

CurlyMoo commented:
I noticed that you didn't move the condition. This is now made useless because of the movement while it should not.

ypflll commented:
Sorry. You mean it should be like this:

```
/* Don't react to signals */
    if(x == -1 && errno == EINTR) {
        x = 0;
    }
    x = poll(&polls, 1, ms);
```

Right?
I don't fully understand the function of the if.

CurlyMoo commented:
Like this:

```
    (void)read(sysFds[pin], &c, 1);
    lseek(sysFds[pin], 0, SEEK_SET);

    x = poll(&polls, 1, ms);

    /* Don't react to signals */
    if(x == -1 && errno == EINTR) {
        x = 0;
    }
```

An interrupt signal also terminates the poll. The poll will then report failure, while the actual reason is not an interrupt failure but a user requested abort. This function splits these two states. A user signaled abortion is allowed, a poll failure isn't. This prevents possible race conditions in the final application.

ypflll commented:
I get it!  Thank you.
I'll modify it.

CurlyMoo commented:
Another wiringPi issue that wiringX fixes...

ypflll committed:
Remove the first false hardware interruption

In radxa.c, function "poll" is called before "lseek" and it will output a false interruption at the beginning even there is no interruptioni source. The same problem should happen on other boards. This is only tested on radxaboard.
